Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+    Write-Host "Account Type: [$($context.Account.Type)]"
+    Write-Host $([Constants]::SingleDashLine)
+        
+    Write-Host "***To add AAD Auth extension on Linux VMSS(s) in a Subscription, Contributor or higher privileges on the VMSS(s) are required.***" -ForegroundColor $([Constants]::MessageType.Warning)
+   
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch all Virtual Machine Scale Set(s)..."
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list to store resource details.
+    $VMSSDetails = @()
+
+    #Separated Linux VMSS(s)
+    $NonCompliantVMSSDetails = @()
+
+    $reqExtPublisher = "Microsoft.Azure.ActiveDirectory"
+    $reqExtensionType = "AADSSHLoginForLinux"
+    $reqExtensionName = "AADSSHLoginForLinux"
+
+    # No file path provided as input to the script. Fetch all Virtual Machine Scale Set(s) in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        try
+        {
+            Write-Host "Fetching all Virtual Machine Scale Set(s) in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+            # Get all Virtual Machine Scale Set(s) in a Subscription
+            $VMSSDetails =  Get-AzVmss -ErrorAction Stop
+
+            # Seperating required properties
+            $VMSSDetails = $VMSSDetails | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                    @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                    @{N='ResourceName';E={$_.Name}},
+                                                                    @{N='OrchestrationMode';E={$_.OrchestrationMode}},
+                                                                    @{N='OsType';E={$_.VirtualMachineProfile.StorageProfile.OsDisk.OsType}},
+                                                                    @{N='Extensions';E={$_.VirtualMachineProfile.ExtensionProfile.Extensions}}
+                                 
+            Write-Host "Separating Virtual Machine Scale Set(s) for which OS Type is 'Windows' and Orchestration mode is 'Uniform' and AAD Extension is not added..."
+            Write-Host $([Constants]::SingleDashLine)
+                                                
+            # Get-AzVmss -ResourceGroupName V-HARARORATESTRG -VMScaleSetName testvmssForBRS1 -InstanceView: $false -UserData
+            $VMSSDetails | ForEach-Object{
+                $VMSS = $_
+                $VMSS | Add-Member -NotePropertyName isExtPresent -NotePropertyValue $false
+                if ($VMSS.OrchestrationMode -eq "Uniform" -and $VMSS.OsType -eq "Linux")
+                {
+                    $VMSS.Extensions | ForEach-Object{
+                        $VMSSExtension= $_
+                        if($VMSSExtension.Type -eq $reqExtensionType -and $VMSSExtension.Publisher -eq $reqExtPublisher)
```

Comment:
Make it uniform, print message in case extension is found.

Reply:
Printed at the end as suggested in other comment.